docs/models/advanced/ibakedmodel.md

```diff
@@ -0,0 +1,48 @@
+
+### `isAmbientOcclusion`
+
+If the model is being rendered as a block in the world, the block in question does not emit any light, and ambient occlusion is enabled, this causes the model to be rendered with ambient occlusion.
+
+### `isGui3D`
+
+If the model is being rendered as an item in an inventory, on the ground as an entity, on an item frame, etc. this makes it look "flat." In GUIs this also disables the lighting.
+
+### `isBuiltInRenderer`
+
+!!! important
+    Unless you know what you're doing and are OK with using deprecated features, just `return false` from this and continue on.
+
+When rendering this as an item, returning `true` causes the model to not be rendered, instead falling back to `TileEntityItemStackRenderer::renderItem`. For certain vanilla items such as chests and banners, this method is hardcoded to copy data from the item into a `TileEntity`, before using a `TileEntitySpecialRenderer` to render that TE in place of the item. For all other items, this queries a registry to find the `Class<? extends TileEntity>` corresponding to the item, finds the `TileEntitySpecialRenderer<T>` for that, and renders that in place of the model itself. `ForgeHooksClient::registerTESRItemStack` is used to register this kind of mapping. When a TESR is called like this, it receives `null` as it's `TileEntity` parameter.
+
+### `getParticleTexture`
+
+Self-explanatory; whatever texture should be used for the particles. For blocks this shows when an entity falls on it, when it breaks, etc. For items this shows when it breaks or when it's eaten.
+
+### <s>`getItemCameraTransforms`</s>
+
+Deprecated in favor of implementing `IPerspectiveAwareModel`. If a model implements that interface this method will not be called, in favor of an `instanceof` check and a call to `IPerspectiveAwareModel::handlePerspective`. Therefore, most `IBakedModel`s that also implement `IPerspectiveAwareModel` should `return ItemCamerTransforms.DEFAULT` here. See [Perspective][].
+
+### `getOverrides`
+
+Returns the [`ItemOverrideList`][ItemOverrideList] to use for this model. This is only used if this model is being rendered as an item.
+
+### `getQuads`
+
+This is the main method of `IBakedModel`. It returns `BakedQuad`s, which contain the low-level vertex data that will be used to render the model. If the model is being rendered as a block, then the `IBlockState` passed in is non-null. Additionally, when applicable, [`Block::getExtendedState`][extended blockstates] is called to create the passed `IBlockState`, which allows for arbitrary data to be passed from the block to the model.
+
+The `EnumFacing` passed in is used for face culling. If the block against the given side of the block being rendered is opaque, then the faces associated with that side are not rendered. If that parameter is `null`, all faces not associated with a side are returned (that will never be culled).
+
+The `long` parameter is a random number.
```

Comment:
That feels too much like an implementation detail. `IBakedModel`s aren't really supposed to know about `World`s, are they?